dev/sg/internal/bk/bk.go

```diff
@@ -161,6 +161,22 @@ func (c *Client) ListArtifactsByBuildNumber(ctx context.Context, pipeline string
 	return artifacts, nil
 }
 
+// ListArtifactsByJob queries the Buildkite API and retrieves all the artifacts for a particular job
+func (c *Client) ListArtifactsByJob(ctx context.Context, pipeline string, buildNumber string, jobID string) ([]buildkite.Artifact, error) {
+	artifacts, _, err := c.bk.Artifacts.ListByJob(BuildkiteOrg, pipeline, buildNumber, jobID, nil)
+	if err != nil {
+		return nil, err
+	}
+	if err != nil {
+		if strings.Contains(err.Error(), "404 Not Found") {
+			return nil, errors.New("no artifacts because no build or job found")
+		}
+		return nil, err
+	}
+
+	return artifacts, nil
+}
```

Comment:
Looks like there is a duplicate `if err != nil` check here that will always return  `nil, err` if there is an error and never reach the second `if err != nil` check
```suggestion
	if err != nil {
		if strings.Contains(err.Error(), "404 Not Found") {
			return nil, errors.New("no artifacts because no build or job found")
		}
		return nil, err
	}

	return artifacts, nil
}
```

Reply:
Ah good catch